runners/spark/src/main/java/org/apache/beam/runners/spark/SparkCommonPipelineOptions.java

```diff
@@ -84,4 +84,11 @@ static void prepareFilesToStage(SparkCommonPipelineOptions options) {
       PipelineResources.prepareFilesForStaging(options);
     }
   }
+
+  @Description(
+      "Enable/disable async output for operators with possibly large output ( such as splittable DoFn )")
+  @Default.Boolean(true)
+  Boolean getEnableAsyncOperatorOutput();
```

Comment:
Is this really what it does? Is it async? My understanding was, that it only uses two threads that pass data using BlockingQueue. Did I miss something?

Reply:
Yes, you are correct. I will think about the better name. However, I wonder, does it make sense to have this parameter at all. When implemented correctly, why would we want to turn it off.

Reply:
It can bring overhead (adding threads), so it might make sense to disable it. We could make it opt-in first, if it looks good, we can do it out-out and then perhaps remove it.

Reply:
Is it possible to make sure validatesRunner executes for both this being true and also false?

Reply:
How about using [ExperimentalOptions](https://github.com/apache/beam/blob/master/sdks/java/core/src/main/java/org/apache/beam/sdk/options/ExperimentalOptions.java#L50-L57) to opt-in initially? Removing a pipeline option again is always a hassle as it obviously is a breaking change.

Reply:
> Is it possible to make sure validatesRunner executes for both this being true and also false?

Running the validatesRunner suite for both cases is a very good point. Thanks for bringing that up @je-ik!
In [validatesRunner](https://github.com/apache/beam/blob/master/runners/spark/spark_runner.gradle#L409-L417) you could trigger [validatesRunnerBatch](https://github.com/apache/beam/blob/master/runners/spark/spark_runner.gradle#L245-L286) for each of the two cases.

Reply:
@JozoVilcek Actually, regarding the comment of @aromanenko-dev. We can limit tests for the new feature / experiment easily to `SplittableDoFnTest`.

Reply:
@mosche can you share an example how do I do that? Alternatively it would make sense to me jsut keep one validates runner with experiment turned on. That means test all iterator on majority of operators but the splittable dofn where we test the experiment. Old iterator does not distinguish what is the operator it is connecting to.

Reply:
That should be fairly simple, just add the following to the test task with the experiment enabled
```
setTestNameIncludePatterns(["org.apache.beam.sdk.transforms.SplittableDoFnTest"])
```

Reply:
I think I got it. Not a category but test class filter. I will make a change